lms/djangoapps/mobile_api/courses/views.py

```diff
@@ -0,0 +1,126 @@
+from rest_framework.response import Response
+from courseware.access import is_mobile_available_for_user
+from student.models import CourseEnrollment
+from social.apps.django_app.default.models import UserSocialAuth
+from mobile_api.users.serializers import CourseEnrollmentSerializer
+from openedx.core.djangoapps.user_api.api.profile import preference_info
+import serializers
+import facebook
+import json
+import urllib2
+
+# TODO: change this to final config
+from ..settings import FB_SETTINGS
+_FACEBOOK_API_VERSION = FB_SETTINGS['_FACEBOOK_API_VERSION']
+
+
+class CoursesWithFriends(generics.ListAPIView):
+    """
+    **Use Case**
+
+        API endpoint for retriving all the courses that a users friends are in. 
+        Note that only friends that allow their courses to be shared will be included.
+
+    **Example request**
+
+        GET /api/mobile/v0.5/courses/friends
+
+    **Response Values**
+
+        [{   "created": "2014-12-09 16:58:31.926438",
+            "mode": "honor",
+            "is_active": "True",
+            "course": {
+                "course_about": "http://testserver/api/mobile/v0.5/course_info/org.4/course_4/Run_4/about",
+                "course_updates": "http://testserver/api/mobile/v0.5/course_info/org.4/course_4/Run_4/updates",
+                "number": "course_4",
+                "org": "org.4",
+                "video_outline": "http://testserver/api/mobile/v0.5/video_outlines/courses/org.4/course_4/Run_4",
+                "id": "org.4/course_4/Run_4",
+                "latest_updates": {
+                    "video": "None"
+                },
+                "end": "None",
+                "name": "Run 4",
+                "course_handouts": "http://testserver/api/mobile/v0.5/course_info/org.4/course_4/Run_4/handouts",
+                "start": "2030-01-01 00:00:00",
+                "course_image": "/c4x/org.4/course_4/asset/images_course_image.jpg"
+                }
+            },
+            ...
+        ]
+    """
+
+    authentication_classes = (OAuth2Authentication, SessionAuthentication)
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = serializers.CoursesWithFriendsSerializer
+
+
+    def list(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.GET, files=request.FILES)
+        if serializer.is_valid():
+            # Get friends from Facebook
+            graph = facebook.GraphAPI(serializer.object['oauth_token'])
+            friends = graph.request(_FACEBOOK_API_VERSION + "me/friends")
+            data = self.get_pagination(friends)
+            # Get the linked edX user if it exists
+            friends_that_are_edX_users = []
+            for friend in data:
+                query_set = UserSocialAuth.objects.filter(uid=unicode(friend['id']))
+                if query_set.count() == 1: 
+                    friend['edX_id'] = query_set[0].user_id
```

Comment:
The edx user id is used to find the course enrolment objects that relate to the user. It is not shared publicly. 
It is referenced here: 
https://github.com/BNOTIONS/edx-001-platform/blob/master/lms/djangoapps/mobile_api/courses/views.py#L97